lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -194,16 +198,44 @@ class HtmlImage implements ui.Image {
         ctx.drawImage(imgElement, 0, 0);
         final DomImageData imageData = ctx.getImageData(0, 0, width, height);
         return Future<ByteData?>.value(imageData.data.buffer.asByteData());
-      default:
+      case ui.ImageByteFormat.png:
+        if (imgElement.src?.startsWith('data:image/png') ?? false) {
+          return _byteDataFromDataString(imgElement.src!);
+        }
+
+        final DomCanvasElement canvas = createDomCanvasElement()
+          ..width = width.toDouble()
+          ..height = height.toDouble();
+        final DomCanvasRenderingContext2D ctx = canvas.context2D;
+        ctx.drawImage(imgElement, 0, 0);
+        // canvas.toDataURL defaults to: 'image/png'
+        final String dataUrl = canvas.toDataURL();
```

Comment:
After reading the image data back, set `canvas.width` and `canvas.height` to zero. Safari on iOS will crash if the GC doesn't collect 2D canvases fast enough. Setting the image size to zero causes Safari to free internal image memory immediately.